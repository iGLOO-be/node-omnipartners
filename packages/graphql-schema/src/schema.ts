import {
  buildSchema as typeGraphqlBuildschema,
  BuildSchemaOptions,
} from "type-graphql";
import { DirectCashbackVoucherListItemResolver } from "./deals/DirectCashbackVoucherListItemResolver";
import { DataAnimalBreedResolver } from "./metadata/DataAnimalBreedResolver";
import { DataAnimalTypeResolver } from "./metadata/DataAnimalTypeResolver";
import { DataAnimalUniverseResolver } from "./metadata/DataAnimalUniverseResolver";
import { DataCountryResolver } from "./metadata/DataCountryResolver";
import { DataLanguageResolver } from "./metadata/DataLanguageResolver";
import { DataLegalFormResolver } from "./metadata/DataLegalFormResolver";
import { DataPlaceOfPurchaseResolver } from "./metadata/DataPlaceOfPurchaseResolver";
import { DataSubscriptionResolver } from "./metadata/DataSubscriptionResolver";
import { DataUserTitleResolver } from "./metadata/DataUserTitleResolver";
import { PartnerListItemResolver } from "./partner/ParnerListItemResolver";
import { PartnerResolver } from "./partner/PartnerResolver";
import { DealResolver } from "./resolvers/DealResolver";
import { ProductResolver } from "./resolvers/ProductResolver";
import { UserAddressCreateResolver } from "./user/UserAddressCreateResolver";
import { UserAddressUpdateResolver } from "./user/UserAddressUpdateResolver";
import { UserCreateResolver } from "./user/UserCreateResolver";
import { UserPartnerRelationCreateResolver } from "./user/UserPartnerRelationCreateResolver";
import { UserPartnerRelationDeleteResolver } from "./user/UserPartnerRelationDeleteResolver";
import { UserPetCreateResolver } from "./user/UserPetCreateResolver";
import { UserPetUpdateResolver } from "./user/UserPetUpdateResolver";
import { UserResolver } from "./user/UserResolver";
import { UserUpdateResolver } from "./user/UserUpdateResolver";

export const buildFullSchema = async (options?: Partial<BuildSchemaOptions>) =>
  typeGraphqlBuildschema({
    resolvers: [
      UserResolver,
      UserCreateResolver,
      UserUpdateResolver,
      UserPetCreateResolver,
      UserPetUpdateResolver,
      UserAddressCreateResolver,
      UserAddressUpdateResolver,
      UserPartnerRelationCreateResolver,
      UserPartnerRelationDeleteResolver,
      DataCountryResolver,
      DataPlaceOfPurchaseResolver,
      DataAnimalBreedResolver,
      DataAnimalTypeResolver,
      DataAnimalUniverseResolver,
      DataLanguageResolver,
      DataSubscriptionResolver,
      DataUserTitleResolver,
      DataLegalFormResolver,
      DealResolver,
      ProductResolver,
      PartnerResolver,
      PartnerListItemResolver,
      DirectCashbackVoucherListItemResolver,
    ],
    ...options,
  });
