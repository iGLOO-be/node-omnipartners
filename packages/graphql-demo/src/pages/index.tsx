import React from "react";
import { CountryList } from "../components/CountryList";

const Homepage = () => <div>
  <h1>Country list:</h1>
  <CountryList />
</div>;

export default Homepage;
