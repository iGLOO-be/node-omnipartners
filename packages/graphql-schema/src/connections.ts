import * as Relay from "graphql-relay";
import { ArgsType, Field, ObjectType } from "type-graphql";

export type ConnectionCursor = Relay.ConnectionCursor;

@ObjectType()
export class PageInfo {
  @Field(() => Boolean)
  public hasNextPage?: boolean | null;
  @Field(of => String, { nullable: true })
  public nextMarker?: ConnectionCursor | null;
}

@ArgsType()
export class ConnectionArgs {
  @Field({ nullable: true, description: "Paginate next marker" })
  public nextMarker?: ConnectionCursor;
  @Field({ nullable: true, description: "Paginate limit" })
  public limit?: number;
}

export {
  connectionFromArray,
  connectionFromPromisedArray,
  connectionFromArraySlice,
  connectionFromPromisedArraySlice,
} from "graphql-relay";
