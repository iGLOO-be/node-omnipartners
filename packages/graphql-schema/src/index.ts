import "reflect-metadata";

export { buildSchema, BuildSchemaOptions } from "type-graphql";

export { DataAnimalBreedResolver } from "./metadata/DataAnimalBreedResolver";
export { DataAnimalTypeResolver } from "./metadata/DataAnimalTypeResolver";
export {
  DataAnimalUniverseResolver,
} from "./metadata/DataAnimalUniverseResolver";
export { DataCountryResolver } from "./metadata/DataCountryResolver";
export { DataLanguageResolver } from "./metadata/DataLanguageResolver";
export { DataLegalFormResolver } from "./metadata/DataLegalFormResolver";
export { DataSubscriptionResolver } from "./metadata/DataSubscriptionResolver";
export { DataUserTitleResolver } from "./metadata/DataUserTitleResolver";
export { DealResolver } from "./resolvers/DealResolver";
export { ProductResolver } from "./resolvers/ProductResolver";
export { Context } from "./types/Context";
export { UserAddressCreateResolver } from "./user/UserAddressCreateResolver";
export { UserAddressUpdateResolver } from "./user/UserAddressUpdateResolver";
export { UserCreateResolver } from "./user/UserCreateResolver";
export { UserPetCreateResolver } from "./user/UserPetCreateResolver";
export { UserPetUpdateResolver } from "./user/UserPetUpdateResolver";
export { UserResolver } from "./user/UserResolver";
export { UserUpdateResolver } from "./user/UserUpdateResolver";

export * from "./schema";
export * from "./utils";
