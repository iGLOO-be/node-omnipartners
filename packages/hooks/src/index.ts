export * from "./user";
export * from "./metadata";
export * from "./partners/useFetchPartnerLocator";
export * from "./partners/usePartnerDetails";
export * from "./deals/useDealCheckSecureCode";
export * from "./deals/useDealSubscribe";
export * from "./lib/useAsyncStorage";
export * from "./lib/dataIdFromObject";
