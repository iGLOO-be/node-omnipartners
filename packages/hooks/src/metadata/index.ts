export * from "./useMetadataUserTitles";
export * from "./useMetadataLegalForms";
export * from "./useMetadataCountries";
export * from "./useMetadataSubscriptions";
export * from "./useMetadataAnimalBreeds";
export * from "./useMetadataAnimalUniverses";
export * from "./useMetadataPartnerTypes";
export * from "./useMetadataPlaceOfPurchase";
export * from "./useMetadataAccountPasswordPolicy";
export * from "./useMetadataAnimalPathologies";
export * from "./useMetadataAnimalLifestyles";
export * from "./useMetadataAnimalSpecialNeeds";
