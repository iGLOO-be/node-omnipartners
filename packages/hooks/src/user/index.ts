export * from "./useUser";
export * from "./UserProvider";
export * from "./useUserEmailExist";
export * from "./useUserLogout";
export * from "./useUserLogin";
export * from "./useUserRecoverPassword";
export * from "./useUserCreate";
export * from "./useUserConfirmLegalForms";
export * from "./address";
export * from "./useUserUpdate";
export * from "./useUserSubscriptions";
export * from "./useUserSubscriptionsUpdate";
export * from "./useUserIsLogged";
export * from "./useUserCheckValidRemoteAuth";
export * from "./pets";
export * from "./tokenContext";
export * from "./children";
export * from "./favourites";
export * from "./segments";
export * from "./useUserLoginByAccessToken";
export * from "./useUserLoginByPublicToken";
export * from "./useUserCheckPasswordTokenValidity";
export * from "./useUserUpdateRecoveredPassword";
export * from "./useUserDeleteSelf";
