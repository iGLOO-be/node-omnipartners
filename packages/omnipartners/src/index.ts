import depd from "depd";

import createLogger from "./lib/logger";

import Data from "./api/data";
import Deals from "./api/deals";
import EventLogger from "./api/eventLogger";
import Identity from "./api/identity";
import Loyalty from "./api/loyalty";
import Metadata from "./api/metadata";
import Partners from "./api/partners";
import Products from "./api/products";
import Api, { IApiOptions } from "./lib/Api";
import { OmnipartnersError } from "./lib/errors";

export { createLogger, OmnipartnersError };

export * from "./types";
export * from "./data-types";
export * from "./partner-types";
export * from "./api/metadata";
export * from "./api/deals";
export * from "./api/partners";
export * from "./api/products";
export * from "./api/loyalty";
export * from "./api/identity";
export * from "./lib/errors";

const deprecate = depd("API");

export interface IOmnipartnersConfig {
  cis?: IApiOptions;
  partners?: IApiOptions;
  products?: IApiOptions;
  loyalty?: IApiOptions;
  deals?: IApiOptions;
  metadata?: IApiOptions;
  eventLogger?: IApiOptions;
}

export enum omnipartnersApiNames {
  identify = "identify",
  identity = "identity",
  partners = "partners",
  loyalty = "loyalty",
  products = "products",
  deals = "deals",
  data = "data",
  metadata = "metadata",
  eventLogger = "eventLogger",
}

export class Omnipartners {
  public identify: Identity;
  public identity: Identity;
  public partners: Partners;
  public loyalty: Loyalty;
  public data: Data;
  public products: Products;
  public deals: Deals;
  public metadata: Metadata;
  public eventLogger: EventLogger;
  private apis: { [k in omnipartnersApiNames]: Api };

  constructor(config: IOmnipartnersConfig) {
    this.identity = new Identity(config.cis);
    this.identify = new Identity(config.cis);
    this.partners = new Partners(config.partners);
    this.loyalty = new Loyalty(config.loyalty);
    this.data = new Data(config.cis);
    this.products = new Products(config.products);
    this.deals = new Deals(config.deals);
    this.metadata = new Metadata(config.metadata);
    this.eventLogger = new EventLogger(config.eventLogger);
    this.apis = {
      identity: this.identity,
      identify: this.identify,
      partners: this.partners,
      loyalty: this.loyalty,
      data: this.data,
      products: this.products,
      deals: this.deals,
      metadata: this.metadata,
      eventLogger: this.eventLogger,
    };

    deprecate.property(
      this,
      "identify",
      "API `identify` is deprecated. Please use `identity`.",
    );

    deprecate.property(
      this,
      "data",
      "API `data` is deprecated. Please use `metadata`.",
    );
  }

  public use(
    fn: (api: Api) => void,
    {
      except,
    }: {
      except?: omnipartnersApiNames[];
    } = {},
  ) {
    Object.keys(omnipartnersApiNames)
      .filter(api => !except || except.indexOf(api as any) < 0)
      .forEach(api => fn((this.apis as any)[api]));
  }
}

export default (config: IOmnipartnersConfig) => {
  return new Omnipartners(config);
};
