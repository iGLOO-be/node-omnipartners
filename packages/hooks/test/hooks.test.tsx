import * as hooks from '../src';

describe('hooks', () => {
  it('exports', () => {
    expect(Object.keys(hooks).sort()).toMatchSnapshot()
  });
});
