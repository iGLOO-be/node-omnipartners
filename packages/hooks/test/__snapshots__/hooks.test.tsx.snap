// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hooks exports 1`] = `
Array [
  "UserAddressFragment",
  "UserAddressQuery",
  "UserChildrenFragment",
  "UserChildrenQuery",
  "UserFavouritesFragment",
  "UserFavouritesQuery",
  "UserPetsFragment",
  "UserPetsQuery",
  "UserProvider",
  "UserSegmentsFragment",
  "UserSegmentsQuery",
  "UserSubscriptionsFragment",
  "UserSubscriptionsQuery",
  "dataIdFromObject",
  "useBetterLazyQuery",
  "useDealCheckSecureCode",
  "useDealSubscribe",
  "useDealVisiblePartnerForUser",
  "useFetchLoyaltyRetrieveBalance",
  "useFetchPartnerLocator",
  "useFetchUser",
  "useFetchUserCheckPasswordTokenValidity",
  "useLogin",
  "useLoyaltyRetrieveBalance",
  "useMetadataAccountPasswordPolicy",
  "useMetadataAnimalBreeds",
  "useMetadataAnimalLifestyles",
  "useMetadataAnimalPathologies",
  "useMetadataAnimalSpecialNeeds",
  "useMetadataAnimalTypes",
  "useMetadataAnimalUniverses",
  "useMetadataCountries",
  "useMetadataLegalForms",
  "useMetadataPartnerTypes",
  "useMetadataPlaceOfPurchase",
  "useMetadataSubscriptions",
  "useMetadataUserTitles",
  "usePartnerDetails",
  "useUser",
  "useUserAddress",
  "useUserAddressCreate",
  "useUserAddressUpdate",
  "useUserCheckPasswordTokenValidity",
  "useUserCheckValidRemoteAuth",
  "useUserChildren",
  "useUserChildrenCreate",
  "useUserChildrenUpdate",
  "useUserConfirmLegalForms",
  "useUserContext",
  "useUserCreate",
  "useUserDeleteSelf",
  "useUserEmailExist",
  "useUserFavourites",
  "useUserFavouritesCreate",
  "useUserFavouritesDelete",
  "useUserIsLogged",
  "useUserLogin",
  "useUserLoginByAccessToken",
  "useUserLoginByPublicToken",
  "useUserLogout",
  "useUserPets",
  "useUserPetsCreate",
  "useUserPetsUpdate",
  "useUserPublicSubscriptions",
  "useUserPublicSubscriptionsUpdate",
  "useUserRecoverPassword",
  "useUserSegments",
  "useUserSubscriptions",
  "useUserSubscriptionsUpdate",
  "useUserToken",
  "useUserUpdate",
  "useUserUpdateRecoveredPassword",
  "useUserUpdateToken",
  "userTokenContext",
]
`;
